tests/unit/remote/ssh/test_ssh.py

```diff
@@ -72,3 +76,47 @@ def test_port(self):
         config["port"] = port
         remote = RemoteSSH(None, config)
         self.assertEqual(remote.port, port)
+
+
+class TestRemoteSSHConnection(TestCase):
+    def setUp(self):
+        super(TestRemoteSSHConnection, self).setUp()
+        self.user = 'user'
+        self.key_path = os.path.join(__here__, "{0}.key".format(self.user))
+        users = {
+            self.user: self.key_path,
+        }
+        self.server = mockssh.Server(users)
+        self.server.__enter__()
```

Comment:
I'm probably missing something, but are you sure your `_client()` is indeed using mockssh's `client`? It looks to me that `remote.ssh` that you are using is trying to connect to the machine itself and not to the mockssh's server.

Reply:
Also, by the looks of it, you've probably wanted to use pytest fixtures here, right? If so, please feel free to do so, we are simply in the middle of migrating from unittest to pytest, so we have a legacy unittest tests and we tend to write new tests with pytest.

Reply:
> I'm probably missing something, but are you sure your `_client()` is indeed using mockssh's `client`?

We intend to fake and mock a SSH server (not client). We use our `client` in `RemoteSSH` to do a test. I don't see any reason why we need to use the mockssh's `client. Could you elaborate why we need to use that?

>It looks to me that `remote.ssh` that you are using is trying to connect to the machine itself and not to the mockssh's server.

No, it connect to mockssh's server. You can see the port that the client tried to connect to.



> Also, by the looks of it, you've probably wanted to use pytest's fixtures here, right? 🙂 If so, please feel free to do so, we are simply in the middle of migrating from unittest to pytest, so we have a legacy unittest tests and we tend to write new tests with pytest.

I'm OK with any option. Just let me know your expectation. There is no problem for me

Reply:
Ah, right. Indeed, the port is not 22. Thanks! :slightly_smiling_face: 

Let's use fixtures for your tests, since it is the most natural approach. Btw, if you are testing SSHConnection itself, let's move your tests to `tests/unit/remote/ssh/test_connection.py`.

Reply:
Sure. I will do it ;)

Reply:
@efiop I updated the point as your expectation